app/commands/solution/generate_readme_file.rb

```diff
@@ -0,0 +1,81 @@
+class Solution
+  class GenerateReadmeFile
+    include Mandate
+
+    initialize_with :solution
+
+    def call
+      [preamble, introduction, instructions, source].compact_blank.join("\n\n")
```

Comment:
I have no idea if think somewhat functional style makes sense, or if you'd prefer a private field that was updated by methods.

Reply:
I'm good with this.

I'd probably format it like this though for readibility:

```suggestion
      [
        preamble, 
        introduction,
        instructions, 
        source
      ].compact_blank.join("\n\n")
```

Bonus points for `compact_blank` - that's new to me! :)

Reply:
`compact_blank` _is_ new.

Reply:
(https://github.com/rails/rails/blob/v6.1.0/activesupport/CHANGELOG.md, since 6.1 ActiveSupport)

Reply:
Modern developers, slowly pulling me away from Rails 3.2 one step at a time!

Reply:
Lol